src/core/crypto.rs

```diff
@@ -74,3 +94,21 @@ pub fn verify_rsa_signature(
         .verify(params, msg, signature)
         .map_err(|_| SignatureVerificationError::CryptoError("bad signature".to_string()))
 }
+
+pub fn verify_ec_signature(
+    key: &CoreJsonWebKey,
+    params: &'static ring_signature::EcdsaVerificationAlgorithm,
+    msg: &[u8],
+    signature: &[u8],
+) -> Result<(), SignatureVerificationError> {
+    let (x, y, crv) = ec_public_key(&key).map_err(SignatureVerificationError::InvalidKey)?;
+    if *crv == CoreJsonCurveType::P521{
+        return Err(SignatureVerificationError::UnsupportedAlg("Only P256 and P384 are supported for now".to_string()));
```

Comment:
I would just make this string `"P521"` since the `Display` output will turn that into `Unsupported signature algorithm: P521` (which is consistent with other instances of this error)